flink-table/flink-table-api-java/src/main/java/org/apache/flink/table/catalog/CatalogManager.java

```diff
@@ -207,28 +207,226 @@ public String getBuiltInDatabaseName() {
+
+		if (currentCatalog != null && currentCatalog.tableExists(objectPath)) {
+			return Optional.of(TableLookupResult.permanent(currentCatalog.getTable(objectPath)));
+		}
+		return Optional.empty();
+	}
+
+	/**
+	 * Retrieves names of all registered catalogs.
+	 *
+	 * @return a set of names of registered catalogs
+	 */
+	public Set<String> listCatalogs() {
+		return Collections.unmodifiableSet(catalogs.keySet());
+	}
+
+	/**
+	 * Returns an array of names of all tables (tables and views, both temporary and permanent)
+	 * registered in the namespace of the current catalog and database.
+	 *
+	 * @return names of all registered tables
+	 */
+	public Set<String> listTables() {
+		return listTables(getCurrentCatalog(), getCurrentDatabase());
+	}
+
+	/**
+	 * Returns an array of names of all tables (tables and views, both temporary and permanent)
+	 * registered in the namespace of the current catalog and database.
+	 *
+	 * @return names of all registered tables
+	 */
+	public Set<String> listTables(String catalogName, String databaseName) {
+		Catalog currentCatalog = catalogs.get(getCurrentCatalog());
+
+		try {
+			return Stream.concat(
+				currentCatalog.listTables(getCurrentDatabase()).stream(),
+				listTemporaryTablesInternal(catalogName, databaseName).map(e -> e.getKey().getObjectName())
+			).collect(Collectors.toSet());
+		} catch (DatabaseNotExistException e) {
+			throw new ValidationException("Current database does not exist", e);
+		}
+	}
+
+	/**
+	 * Returns an array of names of temporary tables registered in the namespace of the current
+	 * catalog and database.
+	 *
+	 * @return names of registered temporary tables
+	 */
+	public Set<String> listTemporaryTables() {
+		return listTemporaryTablesInternal(getCurrentCatalog(), getCurrentDatabase())
+			.map(e -> e.getKey().getObjectName())
+			.collect(Collectors.toSet());
+	}
+
+	/**
+	 * Returns an array of names of temporary views registered in the namespace of the current
+	 * catalog and database.
+	 *
+	 * @return names of registered temporary views
+	 */
+	public Set<String> listTemporaryViews() {
+		return listTemporaryTablesInternal(getCurrentCatalog(), getCurrentDatabase())
+			.filter(e -> e.getValue() instanceof CatalogView)
+			.map(e -> e.getKey().getObjectName())
+			.collect(Collectors.toSet());
+	}
+
+	private Stream<Map.Entry<ObjectIdentifier, CatalogBaseTable>> listTemporaryTablesInternal(
+			String catalogName,
+			String databaseName) {
+		return temporaryTables
+			.entrySet()
+			.stream()
+			.filter(e -> {
+				ObjectIdentifier identifier = e.getKey();
+				return identifier.getCatalogName().equals(catalogName) &&
+					identifier.getDatabaseName().equals(databaseName);
+			});
+	}
+
+	/**
+	 * Lists all available schemas in the root of the catalog manager. It is not equivalent to listing all catalogs
+	 * as it includes also different catalog parts of the temporary objects.
+	 *
+	 * <b>NOTE:</b>It is primarily used for interacting with Calcite's schema.
```

Comment:
mark the following calcite oriented methods `@Internal`?

From Internal.java, "Interface to mark methods within stable, public APIs as an internal developer API."

Reply:
The whole `CatalogManager` class is internal (and annotated appropriately). It makes no sense to additionally mark methods as internal. This is just a note for the flink developers/contributor what it the purpose of this method.